report/src/main/webapp/app/components/compareScreens.directive.js

```diff
@@ -0,0 +1,648 @@
+        return;
+      } else if (wrapper.classList.contains(initedClass)) {
+        console.warn('Screen comparison already has already been initialized');
+        return;
+      } else {
+        wrapper.classList.add(initedClass);
+      }
+
+      var button = document.querySelector('.try-new');
+      button.classList.add('button-disabled');
+
+      var infoMsg = document.createElement('div');
+      infoMsg.innerHTML = '<div class="info-msg">Advanced screen comparison in progress - please wait</div>';
+      var tabContent = document.querySelector('.tab-content');
+      tabContent.appendChild(infoMsg);
+
+      var items = wrapper.querySelectorAll('.layout-compare-item .img-responsive:not(.mask)');
+
+      if (items.length !== 2) {
+        return;
+      }
+
+      prepareMarkup(wrapper);
+
+      var img = {
+        imgA: items[0].src,
+        imgB: items[1].src,
+      };
+
+      var canvas = {
+        canvasWrapper: wrapper.querySelector('.canvas-wrapper'),
+        canvasA: document.createElement('canvas'),
+        canvasB: document.createElement('canvas'),
+      };
+
+      var label = {
+        labelA: wrapper.querySelector('.labelA'),
+        labelB: wrapper.querySelector('.labelB'),
+      };
+
+      var ruler = {
+        rulerA: label.labelA.querySelector('.ruler'),
+        rulerB: label.labelB.querySelector('.ruler'),
+      };
+
+      var mask = {
+        maskA: {
+          top: wrapper.querySelector('.maskAtop'),
+          bot: wrapper.querySelector('.maskAbot'),
+        },
+        maskB: {
+          top: wrapper.querySelector('.maskBtop'),
+          bot: wrapper.querySelector('.maskBbot'),
+        },
+        customMask: wrapper.querySelector('.customMasks'),
+        maskSize: null,
+      };
+
+      var context = {
+        contextA: canvas.canvasA.getContext('2d'),
+        contextB: canvas.canvasB.getContext('2d'),
+      };
+
+      var imgSize = {
+        imgSizeA: null,
+        imgSizeB: null,
+      };
+
+      var simple = {
+        simpleA: null,
+        simpleB: null,
+      };
+
+      var cursor = {
+        cursorA: 0,
+        cursorB: 0,
+        cursorBstart: 0,
+      };
+
+      var difs = {
+        main: wrapper.querySelector('.difs'),
+        difsA: [],
+        difsB: [],
+      };
+
+      var invertedGroups = {
+        invertedGroupsA: [],
+        invertedGroupsB: [],
+      };
+
+      var groups = [];
+      var group = [];
+
+      loadImage(img.imgA, imgSize.imgSizeA, canvas.canvasA, context.contextA, simple.simpleA, function (returnedSimple, returnedImgSize) {
+        simple.simpleA = returnedSimple;
+        imgSize.imgSizeA = returnedImgSize;
+      });
+
+      loadImage(img.imgB, imgSize.imgSizeB, canvas.canvasB, context.contextB, simple.simpleB, function (returnedSimple, returnedImgSize) {
+        simple.simpleB = returnedSimple;
+        imgSize.imgSizeB = returnedImgSize;
+        button.classList.remove('button-disabled');
```

Comment:
Could you please encapsulate logic in the lines below into a function, e.g. `markComparisonFinished` or smth. similar?

Reply:
I've actually left it how it were mainly because I'd need to pass more than 15 parameters to this new function. I might think of something else in the future.